pkg/api/controller.go

```diff
@@ -3006,6 +3012,58 @@ func (c *Controller) GetCurrentUser(w http.ResponseWriter, r *http.Request) {
 	writeResponse(w, http.StatusOK, response)
 }
 
+func (c *Controller) sendResetPasswordEmail(email []string, token string) error {
+	return c.Emailer.SendEmail(email, token, token, []string{})
+}
+
+func (c *Controller) PasswordForgot(w http.ResponseWriter, r *http.Request, body PasswordForgotJSONRequestBody) {
+	user, err := c.Auth.GetUserByEmail(r.Context(), body.Email)
+	if err != nil {
+		writeError(w, http.StatusInternalServerError, err)
+		return
+	}
+	if strings.Compare(*user.Email, body.Email) != 0 {
+		writeError(w, http.StatusNotFound, "No such email")
+		return
+	}
+	secret := c.Auth.SecretStore().SharedSecret()
+	currentTime := time.Now()
+	token, err := GenerateJWT(secret, ResetPasswordAudience, auth.InvalidUserID, body.Email, currentTime, currentTime.Add(tokenExpiryDuration))
```

Comment:
Why use the wrong user ID?  At the very least it should be incorporated into the to-be-signed part of the JWT, if not the returned token itself.  Otherwise if I time things right and we swap emails, I can take control of your account.  "Obviously" people would never do that, but for service accounts this is a valid concern.

Reply:
I am not sure why this is an issue.  AFAIK this was already exposed to users in previous JWTs.

If we *are* worried about this (e.g. because it hints at number of users), then we should simply avoid using consecutive user IDs.

Reply:
Accepted. Set it to the current user.